common/setup/builder.go

```diff
@@ -91,7 +91,11 @@ func (b *AppBuilder) WithBrokerClient() {
 }
 
 func (b *AppBuilder) WithProvisionerClient() {
-	b.provisionerClient = provisioner.NewProvisionerClient(b.cfg.Provisioner.URL, b.cfg.Provisioner.QueryDumping)
+	if b.logger == nil {
+		panic("Logger is not defined")
```

Comment:
Is it really the reason to panic?
Needs discussion.

Reply:
I would leave it as it is should never be the case that logger is not defined.

Reply:
I see, but the thing is that you have method WithLogger, and you allow not to call it.
This building method is optional, so if we require logger to be defined, we should make it mandatory.

Reply:
I have moved logger instance creation to builder constructor to make it implicit, ok? https://github.com/kyma-project/kyma-environment-broker/pull/654/commits/475c13b5e1d4336fabcdf019c0dd1654e88cc98f